fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+    .addIndex(IDX + tableName + "_PLR", PARAMETER_NAME_ID, STR_VALUE_LCASE, LOGICAL_RESOURCE_ID)
+    .addIndex(IDX + tableName + "_RPS", LOGICAL_RESOURCE_ID, PARAMETER_NAME_ID, STR_VALUE)
+    .addIndex(IDX + tableName + "_RPL", LOGICAL_RESOURCE_ID, PARAMETER_NAME_ID, STR_VALUE_LCASE)
+    .addPrimaryKey(PK + tableName, ROW_ID)
+    .setIdentityColumn(ROW_ID, Generated.BY_DEFAULT)
+    .addForeignKeyConstraint(FK + tableName + "_PNID", schemaName, PARAMETER_NAMES, PARAMETER_NAME_ID)
+    .addForeignKeyConstraint(FK + tableName + "_RID", schemaName, logicalResourcesTable, LOGICAL_RESOURCE_ID)
+    .setTablespace(fhirTablespace)
+    .addPrivileges(resourceTablePrivileges)
+    .enableAccessControl(this.sessionVariable)
+    .build(model);
+```
+
+If a new resource is added to the specification, the schema utility automatically provisions it on the next execution of the update schema actions. 
+
+Common reasons to modify the Resource's tables are: 
+- **Specification Change (Version-to-Version changes)**
+    - The Resources are saved as Blobs in the database, and are transparent to the FHIR Version changes. The changes from version-to-version are resilient to field add-remove changes, type changes, and resources additions.  This migration must be done manually.
+    - Resource removals must be done manually. 
+- **Column Attribute Changes (Space, Value Type)**
+    - As the column types, changes are changed in any column, a specific alter statement must be executed on the table, and constraints must be changed or relaxed. This migration must be done manually.
+- **Index - Add or Remove or Update**
+    - As indices are removed from the table definition, the removed indices must be manually dropped for each Resource table. 
+    - As indices are added to the table definition, the version of the table must be updated, and the index must be applied and updated manually.
+- **Constraint Updates** If there are Foreign Key updates, the changes must be applied manually and reflected in the code base. 
+- **Search Parameter Changes (Specification and Tenant)**
+    - If the the parameter type or code is changed, the PARAMETERS_NAME and the corresponding table must be updated to remove references to the removed parameter (based on code). 
+    - If there is a new SearchParameter with `code` newly added to the system, the Search Parameter value entry is only updated when the resource is updated. 
+
+For each of the above changes, the version of the table must be incremented. 
+
+The resource table stores the FHIR resource as a compressed blob, changes to the specification or extensions should only impact Search values. 
+
+The tables have various indices - PrimaryKey, Index and UniqueIndex. These indices are created as part of the Java object - Table.  These 
+
+If a modification is made to a Search Parameter (addition, removal), then the resource uses the existing search parameter values until a new version of the resource is processed.
```

Comment:
combine this with the statement on line 239?

Reply:
done.